src/test/java/io/managed/services/test/kafka/KafkaMgmtAPITest.java

```diff
@@ -218,6 +218,123 @@ public void testCreateTopics() {
         kafkaInstanceApi.createTopic(metricTopicPayload);
     }
 
+    @Test(dependsOnMethods = {"testCreateServiceAccount", "testCreateKafkaInstance"})
+    @SneakyThrows
+    public void testForbiddenToExceedPartitionLimitByTopicCreation() {
+
+        var topicName = "topic-part-exceed-create";
+        final int maxPartitionLimit = KafkaMgmtApiUtils.getPartitionLimitMax(kafkaMgmtApi, kafka);
+        log.info("Max partition limit per given instance: {}", maxPartitionLimit);
+
+        log.info("Attempt to create topic '{}', with too many partitions: {}", topicName, maxPartitionLimit + 1);
+        var payload = new NewTopicInput()
+                .name(topicName)
+                .settings(new TopicSettings().numPartitions(maxPartitionLimit + 1));
+        assertThrows(ApiGenericException.class,
+                () -> kafkaInstanceApi.createTopic(payload));
+    }
+
+    @Test(dependsOnMethods = {"testCreateServiceAccount", "testCreateKafkaInstance"})
+    @SneakyThrows
+    public void testForbiddenToExceedParitionLimitByTopicConfiguration() {
+
+        var topicName = "topic-part-exceed-conf";
+        KafkaInstanceApiUtils.applyTopic(kafkaInstanceApi, topicName);
+
+        final int maxPartitionLimit = KafkaMgmtApiUtils.getPartitionLimitMax(kafkaMgmtApi, kafka);
+        assertThrows(ApiGenericException.class,
+                () -> KafkaInstanceApiUtils.updateTopicPartition(kafkaInstanceApi, topicName, maxPartitionLimit + 1));
+    }
+
+    @Test(dependsOnMethods = {"testCreateServiceAccount", "testCreateKafkaInstance"})
+    @SneakyThrows
+    public void testTotalPartitionLimitByTopicCreation() {
+
+        var topicName = "topic-part-total-exceed-create";
+        final int maxPartitionLimit = KafkaMgmtApiUtils.getPartitionLimitMax(kafkaMgmtApi, kafka);
+        final int currentPartitionCount = KafkaInstanceApiUtils.getPartitionCountTotal(kafkaInstanceApi);
```

Comment:
Will this count include internal partitions (from `__consumer_offsets`, `__redhat_*, `__transaction_state` topics)? If so, I think that would be unexpected behaviour for this test. The limit only applies to non-internal customer partitions.

Reply:
i create an instance, so I have all the topics which do exist offsets,  __redhat_* etc. 
if i call the `getPartitionCountTotal` now it will return 0

the only topic that it will count are thse that user can list (and create) (which are those with regular name or even __ before them assuming user also created them, i.e., not __offset)

Reply:
Great, thanks for the clarification @henryZrncik  :+1: